src/mlpack/core/util/coot_traits.hpp

```diff
@@ -66,6 +66,38 @@ struct IsCube<coot::Cube<eT> >
   static const bool value = true;
 };
 
+// Get the Armadillo equivalent of a Bandicoot type
+
+template<typename MatType>
+struct GetArmaType
+{
+  using type = MatType;
+};
+
+template<typename eT>
+struct GetArmaType<coot::Mat<eT>>
+{
+  using type = arma::Mat<eT>;
+};
+
+template<typename eT>
+struct GetArmaType<coot::Cube<eT>>
+{
+  using type = arma::Cube<eT>;
+};
+
+template<typename eT>
+struct GetArmaType<coot::Row<eT>>
+{
+  using type = arma::Row<eT>;
+};
+
+template<typename eT>
+struct GetArmaType<coot::Col<eT>>
+{
+  using type = arma::Col<eT>;
+};
```

Comment:
I'm not 100% sure, but if this was only used for the `ShuffleData()` variant (which I didn't look at too closely since I think it's not needed anymore), then maybe it makes sense to revert this too.